WooCommerce/WooCommerceTests/ViewRelated/Dashboard/Plugins/SitePluginListViewModelTests.swift

```diff
@@ -174,11 +174,183 @@ class PluginListViewModelTests: XCTestCase {
         // Then
         XCTAssertEqual(viewModel.cellModelForRow(at: IndexPath(row: 0, section: 0)).description, "Lorem ipsum random HTML content")
     }
+
+    func test_section_info_is_correct_after_plugin_status_is_updated() {
+        // Given
+        let activePlugin = SitePlugin.fake().copy(
+            siteID: sampleSiteID,
+            status: .active,
+            name: "AAA"
+        )
+        insert(activePlugin)
+
+        let viewModel = PluginListViewModel(siteID: sampleSiteID, storageManager: storageManager)
+        let expect = expectation(description: "Section info is updated correctly.")
+        viewModel.activate {
```

Comment:
FYI: we have a handy `waitFor()` function to keep these tests and wait times simpler. 

https://github.com/woocommerce/woocommerce-ios/blob/03f86b6656fce7397880fbf99260ecad2d8e4ce8/Yosemite/YosemiteTests/Stores/AddOnGroupStoreTests.swift#L44-L49

Reply:
I don't think `waitFor()` works for this test case though, since I'm triggering `observePlugins()` and expect the `onDataChanged` block to get called after storage is updated. Using `waitFor()` will block the thread so I cannot reach that point where I intend to update storage. But I can certainly use it to test action dispatching! 💯